lighthouse-core/report/handlebar-helpers.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* global Intl */
+
+const Handlebars = require('handlebars');
+const marked = require('marked');
+
+const RATINGS = {
+  GOOD: {label: 'good', minScore: 75},
+  AVERAGE: {label: 'average', minScore: 45},
+  POOR: {label: 'poor'}
+};
+
+function calculateRating(value) {
+  let rating = RATINGS.POOR.label;
+  if (value >= RATINGS.GOOD.minScore) {
+    rating = RATINGS.GOOD.label;
+  } else if (value >= RATINGS.AVERAGE.minScore) {
+    rating = RATINGS.AVERAGE.label;
+  }
+  return rating;
+}
+
+const getTotalScore = aggregation => {
+  return Math.round(aggregation.total * 100);
+};
+
+const getItemRating = value => {
+  if (typeof value === 'boolean') {
+    return value ? RATINGS.GOOD.label : RATINGS.POOR.label;
+  }
+  return calculateRating(value);
+};
+
+const handlebarHelpers = {
+  // Converts a name to a link.
+  'nameToLink': name => {
+    return name.toLowerCase().replace(/\s/g, '-');
+  },
+
+  // Figures out the total score for an aggregation
+  getTotalScore,
+
+  // Converts the total score to a rating that can be used for styling.
+  'getTotalScoreRating': aggregation => {
+    const totalScore = getTotalScore(aggregation);
+    return calculateRating(totalScore);
+  },
+
+  // Converts a value to a rating string, which can be used inside the report
+  // for color styling.
+  getItemRating,
+
+  'shouldShowHelpText': value => (getItemRating(value) !== RATINGS.GOOD.label),
+
+  // Convert numbers to fixed point decimals
+  'decimal': number => {
+    if (number && number.toFixed) {
+      return number.toFixed(2);
+    }
+    return number;
+  },
+
+  // value is boolean?
+  'is-bool': value => (typeof value === 'boolean'),
+
+  // !value
+  'not': value => !value,
+
+  // value === value2?
+  'if_eq': (lhs, rhs, options) => {
```

Comment:
I believe we can't actually use es6 shorthand here since it will change `this` to become the global instead of the applied context of this which is what we want, same goes for `if_not_eq`